simulator/mips/mips_instr.h

```diff
@@ -344,6 +349,21 @@ class MIPSInstr
         }
 
         uint64 get_v_dst() const { return v_dst; }
+        auto get_lwrl_mask() const
+        {
+            uint32 mask = 0x0;
+            uint32 i = 0;
+            while( ( v_imm != (v_dst >> i*8 & v_imm)) && i < 4)
```

Comment:
Let's use `for` for that loop.

```c++
uint32 i = 0;
for (i = 0; i < 4; ++i)
     if ( v_imm == (v_dst >> i*8 & v_imm)))
         break;
```

I think it is much cleaner (though I still do not understand what it does).

If your intention is to demonstrate good knowledge of C, then you have wrong intention. Nobody would blame you for writing not in the "professional" manner, but the skill everyone would appreciate is ability to write code that other can understand.